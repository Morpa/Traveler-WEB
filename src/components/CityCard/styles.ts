import styled, { css } from 'styled-components'

export const Wrapper = styled.a`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    width: 25.1rem;
    border: 1px solid ${theme.colors.blueLow};
    border-radius: 1rem;
    text-decoration: none;
    cursor: pointer;
  `}
`

export const ImageWrapper = styled.div`
  width: 25rem;
  height: 16rem;
  background: #f6f7f8;
  border-radius: 0.9rem 0.9rem 0 0;
  background-image: linear-gradient(
    to right,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );
  background-size: 80rem 14rem;
  animation: placeholderShimmer 1s linear infinite forwards;
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: 0.9rem 0.9rem 0 0;
  }

  @keyframes placeholderShimmer {
    0% {
      background-position: -40rem 0;
    }

    100% {
      background-position: 40rem 0;
    }
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};

    h2 {
      color: ${theme.colors.titleColor};
      margin-bottom: ${theme.spacings.xsmall};
    }

    span {
      font-family: 'Roboto', sans-serif;
      color: ${theme.colors.textColor};
    }
  `}
`
