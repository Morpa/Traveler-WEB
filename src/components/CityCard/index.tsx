import * as S from './styles'

export type CityCardProps = {
  city: string
  imageUrl: string
  locals: number
}

const CityCard = ({ city, imageUrl, locals }: CityCardProps) => (
  <S.Wrapper>
    <S.ImageWrapper>
      <img src={imageUrl} alt={`Picture of ${city}`} />
    </S.ImageWrapper>
    <S.Content>
      <h2>{city}</h2>
      <span>{locals} locais</span>
    </S.Content>
  </S.Wrapper>
)

export default CityCard
