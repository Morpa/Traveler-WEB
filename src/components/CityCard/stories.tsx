import { Story, Meta } from '@storybook/react/types-6-0'
import CityCard, { CityCardProps } from '.'

export default {
  title: 'CityCard',
  component: CityCard,
  args: {
    city: 'Cidade Linda',
    locals: 34,
    imageUrl: 'https://source.unsplash.com/user/willianjusten/'
  }
} as Meta

export const Default: Story<CityCardProps> = (args) => <CityCard {...args} />
