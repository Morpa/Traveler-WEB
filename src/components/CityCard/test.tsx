import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CityCard from '.'

const props = {
  city: 'Cidade Linda',
  locals: 34,
  imageUrl: 'https://source.unsplash.com/user/willianjusten/'
}

describe('<CityCard />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<CityCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.city })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: `Picture of ${props.city}` })
    ).toHaveAttribute('src', props.imageUrl)

    expect(screen.getByText(/locais/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
