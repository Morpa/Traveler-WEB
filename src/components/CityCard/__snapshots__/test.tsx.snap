// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityCard /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25.1rem;
  border: 1px solid #DDE9F0;
  border-radius: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1 {
  width: 25rem;
  height: 16rem;
  background: #f6f7f8;
  border-radius: 0.9rem 0.9rem 0 0;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c1 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 0.9rem 0.9rem 0 0;
}

.c2 {
  padding: 2.4rem;
}

.c2 h2 {
  color: #123952;
  margin-bottom: 1.6rem;
}

.c2 span {
  font-family: 'Roboto',sans-serif;
  color: #617480;
}

<a
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Picture of Cidade Linda"
      src="https://source.unsplash.com/user/willianjusten/"
    />
  </div>
  <div
    class="c2"
  >
    <h2>
      Cidade Linda
    </h2>
    <span>
      34
       locais
    </span>
  </div>
</a>
`
